src/mapgen_valleys.h

```diff
@@ -0,0 +1,202 @@
+/*
+Minetest Valleys C
+Copyright (C) 2010-2015 kwolekr, Ryan Kwolek <kwolekr@minetest.net>
+Copyright (C) 2010-2015 paramat, Matt Gregory
+Copyright (C) 2016 Duane Robertson <duane@duanerobertson.com>
+
+Based on Valleys Mapgen by Gael de Sailly
+ (https://forum.minetest.net/viewtopic.php?f=9&t=11430)
+and mapgen_v7 by kwolekr and paramat.
+
+Licensing changed by permission of Gael de Sailly.
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#ifndef MAPGEN_VALLEYS_HEADER
+#define MAPGEN_VALLEYS_HEADER
+
+#include "mapgen.h"
+
+/////////////////// Mapgen Valleys flags
+#define  MG_VALLEYS_ALT_CHILL     0x01
+#define  MG_VALLEYS_CLIFFS        0x02
+#define  MG_VALLEYS_FAST          0x04
+#define  MG_VALLEYS_HUMID_RIVERS  0x08
+#define  MG_VALLEYS_RUGGED        0x10
+
+class BiomeManager;
+
+extern FlagDesc flagdesc_mapgen_valleys[];
+
+// Global profiler
+//class Profiler;
+//extern Profiler *mapgen_profiler;
+
+
+struct MapgenValleysParams : public MapgenSpecificParams {
+	u32 spflags;
+
+	u16 altitude_chill;
+	s16 cave_water_max_height;
+	s16 humidity;
+	s16 humidity_break_point;
+	s16 lava_max_height;
+	u16 river_depth;
+	u16 river_size;
+	s16 temperature;
+	u16 water_features;
+
+	NoiseParams np_biome_heat;
+	NoiseParams np_biome_heat_blend;
+	NoiseParams np_biome_humidity;
+	NoiseParams np_biome_humidity_blend;
+	NoiseParams np_cliffs;
+	NoiseParams np_corr;
+	NoiseParams np_filler_depth;
+	NoiseParams np_inter_valley_fill;
+	NoiseParams np_inter_valley_slope;
+	NoiseParams np_rivers;
+	NoiseParams np_simple_caves_1;
+	NoiseParams np_simple_caves_2;
+	NoiseParams np_terrain_height;
+	NoiseParams np_valley_depth;
+	NoiseParams np_valley_profile;
+
+	MapgenValleysParams();
+	~MapgenValleysParams() {}
+
+	void readParams(const Settings *settings);
+	void writeParams(Settings *settings) const;
+};
+
+class MapgenValleys : public Mapgen {
+public:
+	EmergeManager *m_emerge;
+	BiomeManager *bmgr;
```

Comment:
Nothing actually calls most of the methods though, so I can't see any reason not to make them private.